src/App.js

```diff
@@ -1,7 +1,32 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import Form from "./components/Form";
-function App() {
+import Todo from "./components/Todo";
+import { nanoid } from "nanoid";
+
+
+function App(props) {
+  const [tasks, setTasks] = useState(props.tasks);
+function addTask(name){
+  let newTask = {
+                id:"todo-" + nanoid(),
+                name: name,
+                complete:false
+                };
+  
+  setTasks([...tasks,newTask]);
+
```

Comment:
Other solutions to avoid mutation of `tasks`
1. 
```
let newArray = Object.assign([], tasks);
newArray.push(newTask);
setTasks(newArray);
```
2.
```
let newArray = tasks.slice();
newArray.push(newTask);
setTasks(newArray);
```